de.metas.contracts/src/main/java/de/metas/contracts/interceptor/C_Flatrate_Term.java

```diff
@@ -569,4 +585,115 @@ public void setMasterDocumentNo(final I_C_Flatrate_Term term)
 			}
 		}
 	}
+	
+	
+	@ModelChange(timings = {
+			ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_AFTER_CHANGE
+	}, ifColumnsChanged = {
+			I_C_Flatrate_Term.COLUMNNAME_ContractStatus, I_C_Flatrate_Term.COLUMNNAME_C_FlatrateTerm_Next_ID
+	})
+	public void updateContractStatusInOrder(final I_C_Flatrate_Term term)
```

Comment:
please move those >100 lines of business logic out of this interceptor

Reply:
PS: further down I saw a dedicated package for contract-orders; whay not mode this MI-method there?